sim/team-validator.js

```diff
@@ -217,6 +217,9 @@ class Validator {
 		if (set.happiness !== undefined && isNaN(set.happiness)) {
 			problems.push(`${name} has an invalid happiness.`);
 		}
+		if (set.hpType && (!dex.getType(set.hpType).exists || ['normal', 'fairy'].includes(toId(set.hpType)))) {
+			problems.push(`${name} has an invalid hidden power type.`);
```

Comment:
You should probably state what the invalid type is. Also "Hidden Power" should be capitalized.